doc/Customization-guide/Endpoint-SDKs/C-SDK/Porting-guide/index.md

```diff
@@ -4,3 +4,69 @@ title: Porting guide
 permalink: /:path/
 sort_idx: 40
 ---
+
+{% include variables.md %}
+
+* TOC
+{:toc}
+
+
+# Prerequisites
+
+Kaa C SDK can be ported to most platforms with OS, without any OS and it's "endian safe".
+
+> **NOTE:**  It supports only 32-bits and 64-bits platforms. A device needs to have a network controller with TCP-IP stack.
+
+# Dependencies
+
+The SDK uses a platform-level layer that depends on the environment where the SDK should run. 
+If you want to port it, you need to modify a few files of the SDK and implement own functions. These files can be divided to the next categories:
+
+## Mandatory		 	
+	
+ - Dynamic memory allocation, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/mem.h). 
+
+ - Networking stack, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/sock.h).
+
+ - Time utilities, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/time.h).
+ 
+ - Output stream, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/stdio.h).
+ 
+ - Default settings, [interface]({{github_url}}client/client-multi/client-c/src/kaa/platform-impl/esp8266/platform/defaults.h).
+  
+> **NOTE:** All references are files which are used by esp8266. You need to create the same but with own implementations.
+ 
+## Optional		 
+
+ - Compile time generated or run time generated keys of the encription. For example, the ESP8266 demo uses the compile time generated key.
```

Comment:
It is not the truth. AES keys which are used for encryption are _always generated in run time_. RSA keys which are used for authentication can be generated either in compile time or run time.